In [11]:
pip install statsmodels


     ---------------------------------------- 0.0/60.4 kB ? eta -:--:--
     ------ --------------------------------- 10.2/60.4 kB ? eta -:--:--
     -------------------------------- ----- 51.2/60.4 kB 525.1 kB/s eta 0:00:01
     -------------------------------- ----- 51.2/60.4 kB 525.1 kB/s eta 0:00:01
     -------------------------------------- 60.4/60.4 kB 321.8 kB/s eta 0:00:00
   ---------------------------------------- 0.0/9.8 MB ? eta -:--:--
   - -------------------------------------- 0.4/9.8 MB 7.6 MB/s eta 0:00:02
   -- ------------------------------------- 0.6/9.8 MB 6.6 MB/s eta 0:00:02
   ---- ----------------------------------- 1.0/9.8 MB 7.3 MB/s eta 0:00:02
   ----- ---------------------------------- 1.5/9.8 MB 7.1 MB/s eta 0:00:02
   ------- -------------------------------- 1.9/9.8 MB 6.7 MB/s eta 0:00:02
   --------- ------------------------------ 2.3/9.8 MB 7.1 MB/s eta 0:00:02
   ---------- ----------------------------- 2.7/9.8 MB 6.9 MB/s eta 0:00:02
   ----------


[notice] A new release of pip is available: 23.3.2 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [12]:
pip install scipy

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.3.2 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [14]:
import numpy as np

# <span style='color:blue'> HYPOTHESIS TESTING

<span style='color:green'>***Hypothesis testing is a fundamental statistical method used to make inferences about population parameters based on sample data. It involves formulating two competing hypotheses: the null hypothesis (H0), which typically states that there is no effect or no difference between groups, and the alternative hypothesis (H1), which asserts the presence of an effect or difference. The process generally entails selecting an appropriate statistical test based on the nature of the data and the research question, calculating a test statistic from the sample data, and determining the probability of obtaining such results under the assumption that the null hypothesis is true, known as the p-value. If the p-value is below a predetermined significance level (usually 0.05), the null hypothesis is rejected in favor of the alternative hypothesis, suggesting evidence for the presence of an effect or difference. Otherwise, if the p-value is above the significance level, the null hypothesis is not rejected. Hypothesis testing provides a systematic framework for drawing conclusions from data, aiding researchers in making informed decisions and drawing valid inferences about the population.***

## <span style='color:blue'> TASKS

#### Q1.Suppose a child psychologist claims that the average time working mothers spend talking to their children is at least 11 minutes per day. You conduct a random sample of 1000 working mothers and find they spend an average of 11.5 minutes per day talking with their children. Assume prior research suggests the population standard deviation is 2.3 minutes.Conduct a test with a level of significance of alpha = 0.05

### <span style='color:blue'>Null Hypothesis
#### <span style='color:indigo'>Ho = &mu; &le; 11 minutes

### <span style='color:blue'>Alternative Hypothesis
#### <span style='color:indigo'>Ha = &mu; > 11 minutes

#### <span style='color:indigo'>x&#772; = 11.5 minutes


#### <span style='color:indigo'>n = 1000

#### <span style='color:indigo'>&sigma; = 2.3 minutes

#### <span style='color:indigo'>&alpha; = 0.05

### Statistical Test
Since we have a sample of 1000, which is greater than 30, and we have a sample mean, standard deviation, and want to compare the sample mean to a population value ,so we acn use Z-test.

rejection criteria is z_statistics>critical value, fail to accept Ho

Formula Z=(x&#772; - &mu;)/(&sigma;/&radic;n)

Where: x&#772; = sample mean (11.5 minutes), &mu; = population mean (11 minutes), &sigma; = population standard deviation (2.3 minutes), n = sample size (1000).

In [8]:
import math

In [10]:
Z = (11.5-11)/(2.3/math.sqrt(1000))
Z

6.874516652539955

### Determine Critical Value or P-Value:
Since we're conducting a one-tailed test (since we're testing if the average time is more than 11 minutes), we will look up the crirical value for a z-score corresponding to an alpha level of 0.05 from z-distribution table. 

#### corresponds to an alpha level of 0.05.
Critical value=1.645

### Conslusion:
### Since our calculated z-score (6.88) is much greater than the critical value (1.645), we reject the null hypothesis. We have sufficient evidence to conclude that the average time working mothers spend talking to their children is indeed more than 11 minutes per day .

### Q2. A coffee shop claims that their average wait time for customers is less than 5 minutes. To test this claim, a sample of 40 customers is taken, and their wait times are recorded. The sample mean wait time is found to be 4.6 minutes with a standard deviation of 0.8 minutes. Perform a hypothesis test at a significance level of 0.05 and determine whether there is enough evidence to support the coffee shop's claim.

### <span style='color:blue'>Null Hypothesis
#### <span style='color:indigo'>Ho = &mu; &ge; 5 minutes

### <span style='color:blue'>Alternative Hypothesis
#### <span style='color:indigo'>Ha = &mu; < 5 minutes

#### <span style='color:indigo'>Sample mean(x&#772;) = 4.6 minutes

#### <span style='color:indigo'>Sample size(n) = 40

#### <span style='color:indigo'>Sample Standard deviation(&sigma;) = 0.8 minutes

#### <span style='color:indigo'>Population mean(&mu;) = 5 minutes

#### <span style='color:indigo'>Significance level(&alpha;) = 0.05(5%)

#### <span style='color:indigo'>Degree of freedom(n-1) = 39

### Statistical Test:
#### Here the population std is unknown, so we can use t-test for this.

##### If the sample size is small (typically <30) and the population standard deviation is unknown, the t-test is appropriate. However, for larger sample sizes (usually greater than 30), the t-test can also be used even if the population standard deviation is unknown due to the Central Limit Theorem, which states that the distribution of sample means approaches a normal distribution as the sample size increases.

### t_test_statistic=(x&#772; - &mu;) / (&sigma; / &radic;n)

In [16]:
t_test_statistic=(4.6 - 5) / (0.8 / np.sqrt(40))
print('t_statistic:',t_test_statistic)

t_statistic: -3.162277660168382



#### The function t.ppf(alpha, df) calculates a critical value from the t-distribution based on the provided significance level (alpha) and degrees of freedom (df).

#### The term "percent point function" (ppf) means it gives you a value from the distribution such that if you randomly select a value from the distribution, there's a certain probability it will be less than or equal to that value.

#### In simpler terms, it helps to find a specific value in the t-distribution where a certain percentage of the distribution falls below that value.

In [18]:
#calculating the critical value with df 39 and alpha 0.05
from scipy.stats import t
critical_value = t.ppf(0.05, 39)

print("Critical value:", critical_value)

Critical value: -1.6848751194974


### Conclusion
#### Since our calculated t-value (-3.16) is less than the critical t-value (-1.685), we fail to accept the null hypothesis.

#### There is enough evidence to support the claim that the average wait time for customers is less than 5 minutes at a significance level of 0.05.